tests/checkers/unittest_non_ascii_name.py

```diff
@@ -0,0 +1,312 @@
+import sys
+from typing import Iterable, Optional
+
+import astroid
+import pytest
+from astroid import nodes
+
+import pylint.checkers.non_ascii_names
+import pylint.interfaces
+import pylint.testutils
+
+
+class TestNonAsciiChecker(pylint.testutils.CheckerTestCase):
+    CHECKER_CLASS = pylint.checkers.non_ascii_names.NonAsciiNamesChecker
+    checker: pylint.checkers.non_ascii_names.NonAsciiNamesChecker
+
+    @pytest.mark.skipif(
+        sys.version_info < (3, 8), reason="requires python3.8 or higher"
+    )
+    def test_kwargs_and_position_only(self):
+        """Even the new position only and keyword only should be found"""
+        node = astroid.extract_node(
+            """
+                def name(
+                    ok,
+                    not_økay,
+                    not_okay_defaułt=None,
+                    /,
+                    p_or_kw_okay=None,
+                    p_or_kw_not_økay=None,
+                    *,
+                    kw_arg_okay,
+                    kw_arg_not_økay,
+                ):
+                    ...
+            """
+        )
+        assert isinstance(node, nodes.FunctionDef)
+        arguments: nodes.Arguments = node.args
+
+        posargs = list(arguments.posonlyargs)
+        args = list(arguments.args)
+        kwargs = list(arguments.kwonlyargs)
+
+        with self.assertAddsMessages(
+            pylint.testutils.MessageTest(
+                msg_id="non-ascii-name",
+                node=posargs[1],
+                args=("Argument", "not_økay"),
+                confidence=pylint.interfaces.HIGH,
+            ),
+            pylint.testutils.MessageTest(
+                msg_id="non-ascii-name",
+                node=posargs[2],
+                args=("Argument", "not_okay_defaułt"),
+                confidence=pylint.interfaces.HIGH,
+            ),
+            pylint.testutils.MessageTest(
+                msg_id="non-ascii-name",
+                node=args[1],
+                args=("Argument", "p_or_kw_not_økay"),
+                confidence=pylint.interfaces.HIGH,
+            ),
+            pylint.testutils.MessageTest(
+                msg_id="non-ascii-name",
+                node=kwargs[1],
+                args=("Argument", "kw_arg_not_økay"),
+                confidence=pylint.interfaces.HIGH,
+            ),
+            ignore_position=True,
+        ):
+            self.checker.visit_functiondef(node)
+
+    @pytest.mark.parametrize(
```

Comment:
I'm probably missing something here, but what's the added benefit of these over the functional tests you wrote?

Reply:
See https://github.com/PyCQA/pylint/pull/5643/files/7c0411a83caccef2059580bbd195ec93ca1b0dcd#r780088668